grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -187,21 +187,30 @@ class UserService {
 
 
 
-
-    def findPersons(String query, Map params) {
-        def criteria = Person.createCriteria()
-        def results = criteria.list (params) {
-            or {
-                ilike("firstName", query)
-                ilike("lastName", query)
-                ilike("email", query)
+    def findPersons(String[] terms) {
+        def results = Person.createCriteria().list {
+
+            if (terms) {
+                terms.each { term ->
+                    and {
+                        or {
+                            ilike("firstName", "%" + term + "%")
+                            ilike("lastName", "%" + term + "%")
+                            ilike("email", "%" + term + "%")
+                        }
+                        or {
+                            ilike("firstName", "%" + term + "%")
```

Comment:
@DominikaHatala  Can you explain this? I think you took my example too literally.

Reply:
@jmiranda  I thought that I had to give or two times because user can add two different terms. But the loop is enough I guess?